### ISSUE TITLE
Bug: [return-await] Gives false positive with unknown values

### ISSUE THREAD

**Issue by danny-pflughoeft** [typescript-eslint/typescript-eslint#9903]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

[link](https://typescript-eslint.io/play/#ts=5.3.3&fileType=.tsx&code=IYZwngdgxgBAZgV2gFwJYHsI2QUxMgCiQGsJ0B3CABWACdgBbALhhLMoEoYBvAKBmy0wPfgJi0cyBLSzBywVMlYRSFanUYBuUQF94qCMAA2R4XzEwomEOiM4AdEfQBzAgHIAkljgHjpmABGTlDEbhy6vDpAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6aRfWaJ2gQwDufSvnRRE0aAHtokcGAC%2BIBUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

### Repro Code

```TypeScript
async function test(unknownParam: unknown) {
  try {
    return await unknownParam;
  } finally {
    console.log('In finally block')
  }
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/return-await": "error",
  },
};
```

### Expected Result

Since `unknownParam` could be a Promise (we don't know), the correct code is `return await unknownParam`, so that the `finally` block will execute after the promise completes, not before.

### Actual Result

The rule does not allow that, and forces you to write incorrect code.

**Comment by kirkwaiblinger** [comment]
TBH my opinion is that we should just remove entirely the check for `return await`ing non-thenables from the rule. It just duplicates the behavior in `require-await`, but with drift in the logic.

https://github.com/typescript-eslint/typescript-eslint/blob/88a8673b185a9ad9d4d7f31059da806f5d9a7449/packages/eslint-plugin/src/rules/return-await.ts#L290-L305

https://github.com/typescript-eslint/typescript-eslint/blob/584db29ec44ce4e9cb71afac35d48994889168e6/packages/eslint-plugin/src/rules/await-thenable.ts#L37-L64

Thoughts? Otherwise we can of course just sync the logic with `await-thenable`.

**Comment by JoshuaKGoldberg** [comment]
Agreed, though that feels like a breaking change either way. 😬

**Comment by kirkwaiblinger** [comment]
Eh. It feels like it doesn't quite technically satisfy any of the requirements in https://typescript-eslint.io/users/versioning#eslint-plugin.

But, it is quietly documented in a single example on [the rule page](https://typescript-eslint.io/rules/return-await) by example only (`invalidInTryCatch6`, `invalidAlways3`). So, arguably would be slightly breaking 🤔 

It sounds like between the choices of 

1. Resolve this ticket by synchronizing the behavior with `await-thenable`, and spinning off separate issue for deprecating and later removing the await-thenable-like check from `return-await`
2. Resolve this ticket by removing the await-thenable-like check from `return-await` entirely
we are
